python/tskit/trees.py

```diff
@@ -8637,8 +8637,11 @@ def pca(
             be automatically generated. Valid random seeds must be between 1 and
             :math:`2^32 − 1`.
         :param np.ndarray range_sketch: Sketch matrix for each window. Default is None.
-        :return: A tuple (U, D, Q) of ndarrays, with the principal component loadings in U
-            and the principal values in D. Q is the range sketch array for each window.
+        :return: A class object with attributes U, D, Q and E. 
```

Comment:
I know this is probably heresy, but could we put names on these variables? Like:
```
@dataclasses.dataclass
class PcaResult:
     """
     The result of a call to TreeSequence.pca() capturing the output values and algorithm convergence details.
     """
     loadings: np.ndarray
     """
     The principle component loadings.
     """
     values: np.ndarray
     """
     The principle component values.
     """
     range_sketch: np.ndarray
     """
     The range sketch. See XXX for details?
     """
     error_bound: np.ndarray
     """
     ...
     """
```
Then, the return would be 
```
:return: An instance of :class:`PcaResult` encapsulating the principle components, loadings and algorithm
    converence details.
```

(A "class object" is something specific, which this isn't)

Reply:
Sounds good. Where should I place the functions and the new class?

Reply:
Put them at the end of trees.py for now. We should probably make a new module for some of this statsy stuff, but let's not bother for now.

Reply:
Note: the functions should not be public, I would think. (the class, yes)

Reply:
Essentially maintenance, yes, but also freedom to change things in the future